velox/exec/tests/SparkAggregationFuzzerTest.cpp

```diff
@@ -48,11 +48,19 @@ int main(int argc, char** argv) {
 
   // TODO: List of the functions that at some point crash or fail and need to
   // be fixed before we can enable.
-  std::unordered_set<std::string> skipFunctions = {
-      "last" // https://github.com/facebookincubator/velox/issues/4482
-  };
+  std::unordered_set<std::string> skipFunctions = {};
+
+  // The results of the following functions depend on the order of input
+  // rows. For some functions, the result can be transformed to a value that
+  // doesn't depend on the order of inputs. If such transformation exists, it
+  // can be specified to be used for results verification. If no transformation
+  // is specified, results are not verified.
+  std::unordered_map<std::string, std::string> orderDependentFunctions = {
+      {"last", ""},
+      {"last_ignore_null", ""},
```

Comment:
Second parameter in last function is a config, usually used in SQL. I think `ignoreNulls` should not appear in RowVectors as a column, so I use different function name to match this behavior. 

Downstream project such as Gluten would maintain a mapping of SparkSQL function to Velox function, if SparkSQL function specify `ignoreNulls`, Gluten should chose correct implementation.

Reply:
For `size` function, https://spark.apache.org/docs/3.3.1/api/sql/#size
> The function returns null for null input if spark.sql.legacy.sizeOfNull is set to false or spark.sql.ansi.enabled is set to true.

`spark.sql.legacy.sizeOfNull` here is static/session function config, no one will use this like `size(expr, legacySizeOfNull)` in plain SQL `select size(expr, config) from tmp`, and Velox provide the ability of use QueryConfig to initialize scalar functions. https://facebookincubator.github.io/velox/develop/scalar-functions.html#access-to-session-properties-and-constant-inputs

But for aggregate function `last`, Velox has no ability to pass QueryConfig-like config for initialization, and ignoreNull used in SQL `select last(expr, ignoreNull) from tmp`, it does not appear in RowVector as column, whether ignoreNull decided by different implementation.

> It would be nice to make be consistent and either remove spark.legacy-size-of-null and replace it with a different flavor of 'size' or add a similar config for first/last aggregate functions.

I agree, but seems design of scalar function and aggregate function are different.

I prefer keep current aggregate function design and provide two version of `last` implementation, Velox is a native library, let downstream project to chose which implementation is needed.

Reply:
@Yohahaha Is the problem then that Aggregate functions do not have access to QueryConfig? Should we fix that?

Reply:
Let me open a discussion for this.